src/content/proxy.js

```diff
@@ -247,14 +247,18 @@ export class Proxy {
 
   // ---------- Quick Add/Exclude Host ---------------------
   static async quickAdd(pref, host) {
-    const activeTab = await this.getActiveTab();
-    const pattern = this.getPattern(activeTab[0].url);
+    const activeTabs = await this.getActiveTab();
+    const activeTabUrlStr = activeTabs[0].url;
+    if (!activeTabUrlStr) { return; }
+    const activeTabUrl = new URL(activeTabUrlStr);
```

Comment:
note: `new URL()` can throw an exception and you're not catching it. In such a case, the rest of the code won't execute.